projects/storefrontlib/src/cms-components/cart/cart-details/cart-details.component.ts

```diff
@@ -37,18 +47,41 @@ export class CartDetailsComponent implements OnInit {
 
   constructor(
     protected cartService: CartService,
-    protected promotionService?: PromotionService
+    protected promotionService?: PromotionService,
+    protected selectiveCartService?: SelectiveCartService,
+    private authService?: AuthService,
+    private routingService?: RoutingService,
+    private featureConfig?: FeatureConfigService
```

Comment:
Those services can be not provided. When you use them first make sure they are defined before accessing properties/methods. Create ticket and add TODO comments to change those dependencies in 2.0 version to be mandatory. Add deprecation comments. (you can see for example in other services/components - there are a lot of them in spartacus currently)

Reply:
Hi @marlass I've created issue #5958 and update the deprecation comments. I've talked with @mszczepaniak the promotionService is also new added service and will be released at 1.5.